### PR TITLE
[ADP-3215] Improve tests for transaction serialization in `read`

### PR DESCRIPTION
**Pull request by HeinrichApfelmus**
This pull request improves the roundtrip tests for transaction serialization / deserialization in the `read` package by

* Putting the hex-encoded golden transactions into a common module `Test.Unit.Cardano.Read.Ledger.Tx`
* Adding a `conwayTx :: Tx Conway`.
* Adding the property `prop_failDowngradeConway` which checks that the serialized `conwayTx` cannot be parsed as a Babbage transaction (because it uses new Conway features).

### Issue Number

ADP-3215

### CODE REVIEW

File: lib/read/test/Cardano/Wallet/Read/Tx/CBORSpec.hs [hunk 1]
```diff
@@ -41,153 +36,38 @@ import Test.QuickCheck
...
+            property $ prop_roundtrip Txs.alonzoTx
+        it "babbage tx" $ do
+            property $ prop_roundtrip Txs.babbageTx
+        it "conway tx" $ do
+            property $ prop_roundtrip Txs.conwayTx
 
-_byronTx :: TxCBOR
-_byronTx = mkTxCBOR Byron ""
+    describe "TxCBOR depends on era" $ do
+        it "fail downgrading Conway to Babbage" $ do
```
Comment (Anviking) [inline lib/read/test/Cardano/Wallet/Read/Tx/CBORSpec.hs]: I believe this doesn't hold in general for Conway -> Babbage?
```suggestion
        it "may fail downgrading Conway to Babbage" $ do
```

Also, I suppose the term "downgrading" could be considered misleading here. In the ledger, a `ByronTxOut` may be _upgraded_ all the way to a `ConwayTxOut` (by the Era translations), but this says nothing about whether or not `deserializeConwayTxOut byronTxOutCbor` will succeed.

Perhaps a more blunt description is better:

```suggestion
        it "may fail deserializing a Conway Tx binary as a Babbage Tx" $ do
```

File: lib/read/test/Cardano/Wallet/Read/Tx/CBORSpec.hs [hunk 2]
```diff
@@ -41,153 +36,38 @@ import Test.QuickCheck
...
-mkTxCBOR _era b = EraValue (K $ unsafeReadBase16 b :: K BL.ByteString era)
+prop_roundtrip :: forall era. IsEra era => Tx era -> Property
+prop_roundtrip tx =
+    parseTxFromCBOR (renderTxToCBOR (EraValue tx))
+        === Right (EraValue tx)
 
-unsafeReadBase16 :: ByteString -> BL.ByteString
-unsafeReadBase16 = either reportError fromStrict . convertFromBase Base16
+prop_failDowngradeConway :: Tx Conway -> Bool
+prop_failDowngradeConway tx =
```
Comment (Anviking) [inline lib/read/test/Cardano/Wallet/Read/Tx/CBORSpec.hs]: Per the other comment, `prop_failDowngradeConway` sounds misleading/overpromising (some ambiguity + believe it doesn't hold for all txs). If it's supposed to be a unit test for demonstrative purposes, why not inline it to the hspec tree?

File: lib/read/test/Cardano/Wallet/Read/Tx/TxIdSpec.hs [hunk 3]
```diff
@@ -47,24 +37,25 @@ import Test.QuickCheck
...
-            prop_matches_TxId maryTx maryTxId
+            prop_matches_TxId Txs.maryTx maryTxId
         it "alonzo tx" $
-            prop_matches_TxId alonzoTx alonzoTxId
+            prop_matches_TxId Txs.alonzoTx alonzoTxId
         it "babbage tx" $
-            prop_matches_TxId babbageTx babbageTxId
+            prop_matches_TxId Txs.babbageTx babbageTxId
+        it "conway tx" $
+            prop_matches_TxId Txs.conwayTx conwayTxId
```
Comment (Anviking) [inline lib/read/test/Cardano/Wallet/Read/Tx/TxIdSpec.hs:58]: 👍 

In the future, probably worth writing in a way that we can't forget to include an era

**Review by Anviking — COMMENTED** [review]
👍 just some concern about how `prop_failDowngradeConway` is presented

File: lib/read/test/Cardano/Wallet/Read/Tx/CBORSpec.hs (hunk 1 above)
Comment (HeinrichApfelmus) [reply lib/read/test/Cardano/Wallet/Read/Tx/CBORSpec.hs]: Thank you, these are excellent points, I completely agree. Changed.

File: lib/read/test/Cardano/Wallet/Read/Tx/CBORSpec.hs (hunk 2 above)
Comment (HeinrichApfelmus) [reply lib/read/test/Cardano/Wallet/Read/Tx/CBORSpec.hs]: 👍  Yes, agreed, thank you, I have inlined it.

File: lib/read/test/Cardano/Wallet/Read/Tx/TxIdSpec.hs (hunk 3 above)
Comment (HeinrichApfelmus) [reply lib/read/test/Cardano/Wallet/Read/Tx/TxIdSpec.hs:58]: When the next hardfork comes — this is good enough for now.

**Review by Anviking — APPROVED** [review]

**Pull request merged**